content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+
+In these cases I always ask myself if it is really worth it, if the time invested in writing these tests will pay off, and if so after how long.
+
+So focusing only on the first of the benefits, for simplicity, we compare the time invested by writing the tests with the gain that will be obtained over time. When investing money, we can understand if the investment is good based on how long it will take to get back into the investment. The same rule applies to software automation.
+
+![enter image description here](https://imgs.xkcd.com/comics/is_it_worth_the_time_2x.png)
+
+To evaluate it, I refer to this scheme found on internet.
+
+## Slow down the CI
+
+Having a test suite makes sense when it is used in a continuous integration system. Typically tests are fairly fast when taken individually, but these can start to become a problem when they are in large quantities.
+
+This may seem like a marginal problem but having a slow CI is problematic. So be careful because at the beginning they can be like grains of sand and when the project becomes very large, the CI can become like a sandstorm and block the development process like a slow build.
+
+## Make harder maintenance and refactoring
+
+This is the case when a change to the codebase (like refactoring an implementation detail or adding a new feature) also implies changes to the test code even if the previous tested behavior doesn't change.
+
+This means that your test code is heavily coupled with your tested code. And can make frustrating, slow and inefficient working on a project.
+
+## Make write worst code
+
+This point is the exact opposite of the one written above on the list of the pros. Indeed to make a portion of code testable we are often forced to restructure our code, but we can harm doing that.
+
+This is called  _"test-induced design damage"_. This is a concept well expressed [here](https://dhh.dk/2014/test-induced-design-damage.html), that explains the process when the pursuit of easier, faster, or more testing warps your code through needless indirection or complexity.
+
+# Many ways of testing (wrong)
+
+Doing automation test is pretty complex. The first time I entered the testing world it was a nightmare, I was overwhelmed by how many concepts are needed only for testing purposes (things like all types of tests, test-driven development (TDD), test doubles, etc). And on top of that all the framework related stuff. So that I asked myself: is it worth it? Short answer: if you want to be a good developer, yes.
+
+So there are many theoretical concepts and many ways of applying them, and it's really very simple to do it the wrong way.
+
+For pure coding related concepts it's more simple to understand what we are doing wrong. Leaving out the functional aspect, because it is obvious that the code that works as we expect is better than the code that does not. Let's analyze it from an architectural point of view.
```

Comment:
With regard to pure coding related concepts, it is easier to understand what we are doing wrong. Leaving out the functional aspect, since it is obvious that the code that works as expected is better than the one that does not, let's analyze it from an architectural point of view.